client/src/components/HomePage.js

```diff
@@ -15,11 +16,12 @@ const HomePage = () => {
   const [activeCategory, setActiveCategory] = useState('All');
 
   useEffect (() => {
-    // get categories from backend and replace mock data
-    setCategories([{icon: 'all_icon', name: 'All'}, {icon: 'breakfast_icon', name: 'Breakfast'}]);
-
-    // get all recipes from backend and replace mock data
-    setRecipes([{id: '1', name: 'Spaghetti bolognese', rating: 3}, {id: '2', name: 'Lentil Soup', rating: 0}]);
+    recipeService.getAll().then(recipes =>
+      setRecipes(recipes)
+    );
+    categoryService.getAll().then(categories =>
+      setCategories(categories)
+    );
```

Comment:
Could a `try catch` structure be implemented here?

Reply:
Modified